src/Providers/Http/HttpTransporter.php

```diff
@@ -66,13 +68,23 @@ public function __construct(
      * {@inheritDoc}
      *
      * @since 0.1.0
+     * @since n.e.x.t Added optional RequestOptions parameter and ClientWithOptions support.
      */
-    public function send(Request $request): Response
+    public function send(Request $request, ?RequestOptions $options = null): Response
     {
         $psr7Request = $this->convertToPsr7Request($request);
 
+        // Merge request options with parameter options, with parameter options taking precedence
+        $mergedOptions = $this->mergeOptions($request->getOptions(), $options);
+
         try {
-            $psr7Response = $this->client->sendRequest($psr7Request);
+            if ($this->client instanceof ClientWithOptionsInterface) {
+                $psr7Response = $this->client->sendRequestWithOptions($psr7Request, $mergedOptions);
+            } elseif ($this->isGuzzleClient($this->client)) {
```

Comment:
Nit-pick: We could change these two clauses to only be relevant if `$mergedOptions !== null`. Not a must have by any means though.

Reply:
Sure! Makes sense. Resolved in 0da66e6